public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -198,11 +197,11 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
   const handleStartAdJob = async (detectorId: string) => {
     try {
       await dispatch(startDetector(detectorId));
-      toastNotifications.addSuccess(
+      core.notifications.toasts.addSuccess(
```

Comment:
Can we use `const { httpClient, notifications } = this.props;` to get notifications like alerting Kibana change? The import path of `CoreStart` looks very long `'../../../../../../src/core/public'`

Reply:
So `httpClient` and `notifications` are core services, and alerting plugin decided to pass these through `props` since the services didn't have to be passed deep down into the component tree. For AD, we have much more places where these services need to be accessed deeply, and is why I decided to use React context to access them instead. See [here](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#how-can-i-avoid-passing-core-services-deeply-within-my-ui-component-tree) for more info

Reply:
Regarding the import path, I think there is a way to shorten & customize import paths in the `tsconfig.json` file, although I haven't looked into it much. I think this would be a good thing to look into later, and could change for other common imports as well, rather than having so many of these long relative imports. Would help prevent imports from breaking during refactoring as well.

Reply:
I think Yaliang agrees with using "context" to access the core services, as in the part of Anomaly Detector feature in Alerting plugin (https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin/blob/master/public/pages/CreateMonitor/containers/AnomalyDetectors/AnomalyDetectors.js#L24), React context is used instead of "props" 😉. And I think the reason that the import path is not outstanding in alerting plugin is it doesn't needs to declare the type there. 
Anyway, it's good to know that there is a way to shorten the import paths.

Reply:
Cool, not a big problem, let's keep as is currently.